src/Tribe/Admin/Notices.php

```diff
@@ -325,16 +334,106 @@ public function has_user_dimissed( $slug, $user_id = null ) {
 			return false;
 		}
 
+		$notice = $this->get( $slug );
+		if (
+			is_object( $notice )
+			&& $notice->recurring
+			&& $this->should_recurring_notice_show( $slug, $user_id )
+		) {
+			return false;
+		}
+
 		return true;
 	}
 
+	/**
+	 * Checks if a given user has dismissed a given notice.
+	 *
+	 * @since      4.3
+	 * @deprecated TBD Deprecated in favor of correcting the typo.
+	 *
+	 * @param string   $slug    The Name of the Notice
+	 * @param int|null $user_id The user ID
+	 *
+	 * @return boolean
+	 */
+	public function has_user_dimissed( $slug, $user_id ) {
+		return $this->has_user_dismissed( $slug, $user_id );
+	}
+
+	/**
+	 * Gets the last Dismissal for a given notice slug and user.
+	 *
+	 * @since TBD
+	 *
+	 * @param string   $slug    Slug of the notice to look for.
+	 * @param int|null $user_id Which user? If null will default to current user.
+	 *
+	 * @return false|\Tribe\Utils\Date_I18n
+	 */
+	public function get_last_dismissal( $slug, $user_id = null ) {
+		if ( is_null( $user_id ) ) {
+			$user_id = get_current_user_id();
+		}
+
+		$dismissed_time = get_user_meta( $user_id, static::$meta_key_time_prefix . $slug, true );
+
+		if ( ! is_numeric( $dismissed_time ) ) {
+			return false;
+		}
+
+		return Dates::build_date_object( $dismissed_time );
+	}
+
+	/**
+	 * Determines if a given notice needs to be re-displayed in case of recurring notice.
+	 *
+	 * @since TBD
+	 *
+	 * @param string   $slug    Slug of the notice to look for.
+	 * @param int|null $user_id Which user? If null will default to current user.
+	 *
+	 * @return false|\Tribe\Utils\Date_I18n
+	 */
+	public function should_recurring_notice_show( $slug, $user_id = null ) {
+		$notice = $this->get( $slug );
+		if ( ! is_object( $notice ) ) {
+			return false;
+		}
+
+		if ( ! $notice->recurring || ! $notice->recurring_interval ) {
+			return false;
+		}
+
+		if ( is_null( $user_id ) ) {
+			$user_id = get_current_user_id();
+		}
+
+		$interval       = Dates::interval( $notice->recurring_interval );
+		$last_dismissal = $this->get_last_dismissal( $slug, $user_id );
+		if ( ! $last_dismissal ) {
+			return false;
```

Comment:
Should this one be `true` (If there is no last dismissal, then it should show?) or am I reading this wrong?

Reply:
ah, gotcha!